### DISCUSSION TITLE
Change GenIdeaModule#ideaCompileOutput to match module localClasspath

### DISCUSSION THREAD

**Discussion by yankun1992** [com-lihaoyi/mill#2238] [Ideas]
I have two module:
```scala
object mio extends ScalaModule {
  override def scalaVersion = "3.2.1"

  override def moduleDeps: scala.Seq[JavaModule] = scala.Seq(mionative)
}

object mionative extends JavaModule {
  // define target to generate native jni library
  def compileNative : T[PathRef] = ???

  override def localClasspath: T[Seq[PathRef]] = T {
    super.localClasspath() :+ compileNative()
  }

}
```

When use 
```bash 
mill mio.run
```
it can load native library succeed but fail use idea run because 
```bash 
mill mill.scalalib.GenIdea/idea
```
use `ideaCompileOutput` to generate module classpath and it fix 
```scala
  def ideaCompileOutput: T[PathRef] = T.persistent {
    PathRef(T.dest / "classes")
  }
```
Maybe we can change the return type of `ideaCompileOutput` to `T[Seq[PathRef]]` or change `GenIdeaImpl#xmlFileLayout` to integrate module `localClasspath` ?

**Comment by lefou** [comment]
`GenIdeaModule.ideaCompileOutput` is the path used by IDEA to write compilation output (class files) to. This content of this path is meant to be managed by IDEA.

Also, changing `GenIdeaImpl.xmlFileLayout` is not helpful, as we only generate config files and we can only use what Idea already supports. We especially have no way to trigger dedicated Mill targets, like your `compileNative` from within IDEA. 

That being said, if your `compileNative` is mostly stable and does not need to re-run often (especially not more often than you re-generate the IDEA project), you could consider to return the `compileNative` result from `resources` instead of `localClasspath`. It still ends in `localClasspath`, but also IDEA will see it.

**Reply by yankun1992** [reply]
Thank you very much, it works!

**Comment by yankun1992** [comment]
By the way, I have developed a mill plugin to compile the jni code implemented by rust. The repository is  [ mill-rust-jni](https://github.com/otavia-projects/mill-rust-jni). Wellcome to use!

**Reply by lefou** [reply]
Nice. Feel free to add your plugin the our documentation: https://com-lihaoyi.github.io/mill/mill/Thirdparty_Plugins.html

Also, as major Mill versions (e.g. `0.10.x` or `0.9.x`) are not binary compatible, it's a convention to add the Mill binary platform as a suffix to the published artifact name, e.g. `_mill0.10`. You can find details here: https://com-lihaoyi.github.io/mill/mill/Using_Plugins.html
